### PR TITLE
Use room name as room intro

### PR DESCRIPTION
**Pull request by weeman1337**
## Checklist

* [x] Tests written for new code (and old code if feasible)
* [x] Linter and other CI checks pass
* [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

Notes: Description of DM room with more than two other people is now being displayed correctly

fixes https://github.com/vector-im/element-web/issues/23094

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Description of DM room with more than two other people is now being displayed correctly ([\#9231](https://github.com/matrix-org/matrix-react-sdk/pull/9231)). Fixes vector-im/element-web#23094.

### CODE REVIEW

File: src/components/views/rooms/NewRoomIntro.tsx [hunk 1]
```diff
@@ -55,19 +55,21 @@ const NewRoomIntro = () => {
         ? room.targets[0]?.userId
         : DMRoomMap.shared().getUserIdForRoomId(roomId);
 
-    let body;
+    let body: JSX.Element;
     if (dmPartner) {
         let introMessage = _t("This is the beginning of your direct message history with <displayName/>.");
-        let caption;
+        let captionElement: JSX.Element = <></>;
```
Comment (t3chguy) [inline src/components/views/rooms/NewRoomIntro.tsx:61]: We don't ever initialise to empty fragments, shouldn't we stay self-consistent, e.g. with L58 for example

Comment (weeman1337) [reply src/components/views/rooms/NewRoomIntro.tsx:61]: If not initialising it, tsc in strict mode complains about using `captionElement` before assignment here https://github.com/matrix-org/matrix-react-sdk/pull/9231/files/e4c4b73b814288ba29a3f95605a6eb4830721d07#diff-c509de2dd1c9eb2add4073f662ec42078874b71b98d13e34d141abff5bb65f32L90 .

Comment (t3chguy) [reply src/components/views/rooms/NewRoomIntro.tsx:61]: null is valid for rendering, but you should be consistent with both vars either way

**Review by t3chguy — APPROVED** [review]

**Comment by grimhilt** [comment]
I think ``tsc.txt`` is unwanted.

**Pull request merged**